### DISCUSSION TITLE
[V3] Modelable properties and file uploads

### DISCUSSION THREAD

**Discussion by esadewater** [livewire/livewire#6203] [7. Bugs (archived)]
### Livewire version

v3.0.0-beta.6

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

1. Create a component A with the HasFileUploads trait and a modelable property which you use for a file upload
2. Create a component B in which you use component A and leverage it's modelable property
3. Upload a file with component A and e.g. dd it in component B. Instead of a TemporaryUploadedFile you will get a string like "livewire-file:...ID....webp" You can still manually rehydrate the property with `TemporaryUploadedFile::unserializeFromLivewireRequest(...)`, but I don't think this is the desired behavior.

### Please confirm (incomplete submissions will not be addressed)

- [x] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@esadewater temporary uploaded files can't be dispatched, hence why you're having an issue dispatching it to the parent. You'd be better off storing the file in the component and then dispatching it's location for use in the parent.

**Reply by esadewater** [reply]
Hey @joshhanley , thanks for your response. Maybe my explanation was a bit misleading. I will add some code to make it a bit clearer.

I have an image input like this (which will display a preview of stored/uploaded file + some extra functionality)
```php
<?php

namespace App\Livewire;

use Livewire\Attributes\Modelable;
use Livewire\Component;
use Livewire\Features\SupportFileUploads\TemporaryUploadedFile;
use Livewire\WithFileUploads;

class ImageInput extends Component
{
    use WithFileUploads;

    /** @var array{file?: ?TemporaryUploadedFile, media?: ?string, clear?: ?bool} $image */
    #[Modelable]
    public array $image = [];

    ...

    /**
     * !!! Only for debugging !!!
     */
    public function updatedImage()
    {
        dd($this->image);
    }
}
```

And I want to use it in another component, e.g. a form to edit a user profile:
```html
<form wire:submit"store">
    <livewire:image-input wire:model="profile_image" />
</form>
```

```php
class ProfileImageForm extends Component
{
    public array $profile_image;

    ...

    public function store()
    {
        dd($this->profile_image);

        ...
    }
}
```

The `dd` in the ImageInput will give me the expected value with a TemporaryUploadedFile:
```
array:3 [▼ // app/Livewire/ImageInput.php:72
  "media" => 75
  "clear" => false
  "file" => Livewire\Features\SupportFileUploads\TemporaryUploadedFile {#1998 ▼
    ...
  }
]
```

But the through `#[Modelable]` updated value in the ProfileImageForm will only give me this:
```
array:3 [▼ // app/Livewire/Web/Account/ProfileImageForm.php:37
  "media" => 75
  "clear" => false
  "file" => "livewire-file:VpGMww9oSGCkyvzDVT9mjcFh0qdl4g-metaWmlDbEpmLTE5MjB3LmpwZw==-.jpg"
]
```

The string `livewire-file:VpGMww9oSGCkyvzDVT9mjcFh0qdl4g-metaWmlDbEpmLTE5MjB3LmpwZw==-.jpg` can manually be converted to a TemporaryUploadedFile by using `TemporaryUploadedFile::unserializeFromLivewireRequest`, but I would expect Livewire to handle this.

**Reply by bebo925** [reply]
I am experiencing this issue also.  I worked through the FileUploadSynth and it all seems to be working as expected.  It seems that something is getting lost through the Modelable attribute, but I didn't get very far diving through that code. I don't see any evidence that Modelable honors the Synth process.

**Reply by Rikaelus** [reply]
> I am experiencing this issue also. I worked through the FileUploadSynth and it all seems to be working as expected. It seems that something is getting lost through the Modelable attribute, but I didn't get very far diving through that code. I don't see any evidence that Modelable honors the Synth process.

A year later and I'm thinking the same thing. I have a Synth set up for a small custom class _(just 3 properties -- cleaner than using an array)_. I see the Synth dehydrating/hydrating successfully in some areas, but I believe Livewire's ignoring the Synth when putting the value into a Modelable property; I'm getting an error about an attempt to assign an array to my component's property that's expecting an object of my class.

Any time I try to push Modelable beyond very simple types I end up in worlds of pain...
